README.md

```diff
@@ -189,12 +201,14 @@ See [C++ Programmer's Toolbox](#c-programmers-toolbox) for details.
     }
     ```
     
+    Also we could return a smart pointer for complex type, such as unique_ptr<std::vector<int>>.
```

Comment:
In this particular case return unique_ptr<std::vector<int>> is unnecessary and this involves copy elision and/or moving semantics which is the same point as stated in this paragraph :)

It is good, however, to add an example demonstrating returning `unique_ptr<SomeType>` where `SomeType` is not movable.